telegram/ext/filters.py

```diff
@@ -603,6 +603,45 @@ def filter(self, message: Message) -> bool:
         xml = mime_type('application/xml')
         zip = mime_type('application/zip')
 
+        class file_extension(MessageFilter):
+            """This filter filters documents by their file ending/extension.
+
+            Note:
+                This Filter only filters by the file ending/extension of the document,
+                    it doesn't check the validity of document.
+                The user can manipulate the file extension of a document and
+                    send media with wrong types that don't fit to this handler.
+
+            Example:
+                ``Filters.document.file_extension('jpg')`` filters files with extension ``.jpg``.
+            """
+
+            file_extension: Optional[List[str]] = None
+
+            def __init__(self, file_extension: Optional[str]):
+                """Initialize the extension you want to filter.
+
+                Args:
+                    file_extension (str, optional): media file extension you want to filter.
+                """
+                if file_extension is not None:
+                    file_extension = file_extension.lstrip(".")
```

Comment:
I do see why you want the user to not have to care about the dot, but I'd prefer to not fiddle with it and instead just document if the dot must be passed or not. Image someone for some reason wanting to filter for a `......ext` extension. That wouldn't work in the current setting.

Reply:
This is the case i didn't think of. Okay.